UnityProject/Assets/Scripts/Botany/Loom.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using Mirror;
+using Random = UnityEngine.Random;
+
+public class Loom : NetworkBehaviour, ICheckedInteractable<HandApply>
+{
+	[Tooltip("How many bundles of raw cotton/durathread to consume.")]
+	[SerializeField]
+	private int bundles = 4;
+
+	[Tooltip("How many sheets of fabric to get.")]
+	[SerializeField]
+	private int sheets = 1;
+	
+	[Tooltip("When raw cotton bundles are processed, this will be created.")]
+	[SerializeField]
+	private GameObject cottonSheet;
+
+	[Tooltip("When raw durathread bundles are processed, this will be created.")]
+	[SerializeField]
+	private GameObject durathreadSheet;
+
+	public bool WillInteract(HandApply interaction, NetworkSide side)
+	{
+		//start with the default HandApply WillInteract logic.
+		if (!DefaultWillInteract.Default(interaction, side)) return false;
+
+		GameObject ObjectInHand = interaction.HandObject;
```

Comment:
This local variable isnt really needed.

Reply:
Actually it is fine here.
It's used on line `32` and `33`
It's saves a bit of time by caching it here.

Reply:
Eh i guess.

Reply:
Should I mark this as resolved?

Reply:
yes, good